primitives/session-manager/src/pallet_session_support.rs

```diff
@@ -0,0 +1,108 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use core::marker::PhantomData;
+use derive_new::new;
+use frame_system::pallet_prelude::BlockNumberFor;
+use log::debug;
+use sp_staking::SessionIndex;
+use sp_std::vec::Vec;
+
+/// Implements [`pallet_session::SessionManager`] and [`pallet_session::ShouldEndSession`] integrated with [`pallet_session_validator_management`].
+///
+/// To use it, wire it in runtime configuration of [`pallet_session`].
+#[allow(dead_code)]
+#[derive(new)]
+pub struct PalletSessionSupport<T> {
+	_phantom: PhantomData<T>,
+}
+
+impl<T: pallet_session_validator_management::Config + pallet_session::Config>
```

Comment:
I would move those trait bounds directly to the type definition above. `PalletSessionSupport` serves no purpose without those traits defined, and the compiler will then produce much better error messages when they are missing.

Reply:
This required adding all bounds to the ShouldEndSession, keeping them here and required some additional code in the test, so I will keep it here, like for the other SessionManager in this crate.